### ISSUE TITLE
getcells() error

### ISSUE THREAD

**Issue by ableitner** [ropensci/seaaroundus#15]
I am having problems with getcells()
The example call works great:
>test<- getcells("POLYGON ((-48.177685950413291 15.842380165289299,
 -54.964876033057919 28.964280991735578,
 -35.960743801652967 27.606842975206646,
 -48.177685950413291 15.842380165289299))")

BUT when I replace those coordinates with different ones, I get this error:
>test2<- getcells("POLYGON ((-47.96675 -60.44316,
  -47.94088 -61.34051,
  -49.80912 -61.34051,
  -49.78325 -60.44316))")
Error in postapi(paste(getapibaseurl(), "spatial/r/shape", sep = "/"),  : 
  Internal Server Error (HTTP 500).

What is going on here? There is no information in the getcells() description, so I am stuck. Everything is the exact same format as the example code.

Also, does anyone know how to use getcells() in a form that can be automated?
If I want to extract hundreds of small boxes I will want to automate this step somehow, using entries in a dataframe as my polygon corners. 
All of my attempts have been foiled- likely because I do not know what the "POLYGON (())" is. 

For example:

>SMinfo <- data.frame(UEcornerLong=-47.96675 ,  UEcornerLat=-60.44316, LEcornerLong=-47.94088 ,  LEcornerLat=-61.34051 , LWcornerLong=-49.80912 , LWcornerLat=-61.34051 , UWcornerLong=-49.78325 ,UWcornerLat=-60.44316 )

>test4<- getcells("POLYGON (( SMinfo$UEcornerLong[1] SMinfo$UEcornerLat[1], SMinfo$LEcornerLong[1] SMinfo$LEcornerLat[1], SMinfo$LWcornerLong[1] SMinfo$LWcornerLat[1], SMinfo$UWcornerLong[1] SMinfo$UWcornerLat[1] ))")
 > Error in postapi(paste(getapibaseurl(), "spatial/r/shape", sep = "/"),  : Internal Server Error (HTTP 500).

OR:
>a<-paste(SMinfo$UEcornerLong[1], SMinfo$UEcornerLat[1], sep=' ')
>b<-paste(SMinfo$LEcornerLong[1], SMinfo$LEcornerLat[1],sep=" ")
>c<-paste(SMinfo$LWcornerLong[1], SMinfo$LWcornerLat[1], sep=' ')
>d<-paste(SMinfo$UWcornerLong[1], SMinfo$UWcornerLat[1])

>getcells(paste("POLYGON ((", paste(a,b,c,d, sep=",\n "), "))" , sep=''))
  Error in postapi(paste(getapibaseurl(), "spatial/r/shape", sep = "/"),  : Internal Server Error (HTTP 500).

Why doesn't the above work? paste("POLYGON ((", paste(a,b,c,d, sep=",\n "), "))" , sep='') returns the exact same thing as the argument inside the example function:

> "POLYGON ((-48.177685950413291 15.842380165289299,
  -54.964876033057919 28.964280991735578,
  -35.960743801652967 27.606842975206646,
  -48.177685950413291 15.842380165289299))"
[1] "POLYGON ((-48.177685950413291 15.842380165289299,\n -54.964876033057919 28.964280991735578,\n -35.960743801652967 27.606842975206646,\n -48.177685950413291 15.842380165289299))"

Any help on this would be greatly appreciated.

**Comment by ableitner** [comment]
Interesting update:
Even when I just slightly modify the example code by deleting a few digits I get an API error:
> test<- getcells("POLYGON ((-48.177413291 15.8423899,
+ -54.96487657919 28.9642809978,
+ -35.9607438967 27.6068429,
+ -48.1776859 15.84238))")
Error in postapi(paste(getapibaseurl(), "spatial/r/shape", sep = "/"),  : 
  Internal Server Error (HTTP 500).

**Comment by sckott** [comment]
@ableitner should be back up, try again now

**Comment by ableitner** [comment]
@sckott still doesn't work. I get the same error message:
getcells("POLYGON ((-48.177685950413291 15.842380165289299,
+ -48.177685950413291 15.842380165289299,
+ -54.964876033057919 28.964280991735578,
+ -35.960743801652967 27.606842975206646,
+ -48.177685950413291 15.8423801652892))")
Error in postapi(paste(getapibaseurl(), "spatial/r/shape", sep = "/"),  : 
  Internal Server Error (HTTP 500).

Though the example works, if I delete 2 digits at the end of the last latitude (above), I get an error.

**Comment by ableitner** [comment]
@sckott Is there a work around option like a downloadable CSV of the cells spatial data that I could download and query directly in R?

**Comment by sckott** [comment]
@ableitner wrt bulk downloads - we (ropensci) can't do that - that is up to Seaaroundus - we can only answer questions about the R package. 

For your query problem - the problem appears to be that your WKT polygon isn't closed correctly - that is, the coordinates of the first point and the last have to be exactly the same. - And the server side code for Seaaroundus is failing when they don't match, try this:

```r
getcells("POLYGON ((-49.78325 -60.44316,
        -47.94088 -61.34051,
        -49.80912 -61.34051,
        -49.78325 -60.44316
    ))"
)
```

or this:

```r
getcells(
  "POLYGON ((-48.177685950413291 15.842380165289299, 
             -48.177685950413291 15.842380165289299, 
             -54.964876033057919 28.964280991735578, 
             -35.960743801652967 27.606842975206646, 
             -48.177685950413291 15.842380165289299
  ))"
)
```

**Comment by ableitner** [comment]
GREAT! This works. I am feeling foolish that I simply was not closing the polygon.

Also, if someone is interested the function CAN be looped:

a<-paste(SMinfo$UEcornerLong[i], SMinfo$UEcornerLat[i], sep=' ')
b<-paste(SMinfo$LEcornerLong[i], SMinfo$LEcornerLat[i],sep=' ')
c<-paste(SMinfo$LWcornerLong[i], SMinfo$LWcornerLat[i], sep=' ')
d<-paste(SMinfo$UWcornerLong[i], SMinfo$UWcornerLat[i], sep=' ')
cellID <- getcells(paste("POLYGON ((", paste(a,b,c,d,a, sep=", "), "))" , sep=""))

**Comment by sckott** [comment]
In this pkg we can try to check the WKT - will try that

**Comment by sckott** [comment]
@ableitner okay, reinstall - and try bad WKT - should error correctly now

**Comment by ableitner** [comment]
Both getcells() and getcelldata() are working great now!

**Comment by sckott** [comment]
great

**Comment by jafflerbach** [comment]
ok not sure what I'm missing here since I've ensured that the first and last points are matching:

```r 
saup_cells <- getcells("POLYGON ((-180 90,-180 -90, 180 -90, 180 90, -180 90))")
Error: The WKT object has a different orientation from the default
```

**Comment by sckott** [comment]
has to do with the winding order I believe of the points in the WKT - comes from our pkg `wicket` for WKT parsing/validating - i have an open issue there https://github.com/ropensci/wicket/issues/19 

this should work `getcells("POLYGON ((-180 90, 180 90, 180 -90, -180 -90, -180 90))")`

they are essentially the same except the order in which the points are drawn is different - i'll work on a way to warn about this and/or try to fix if possible 👉  #17
